src/components/AddForm/index.jsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { Input, InputLabel, Button } from '@mui/material';
+import { createLeader } from 'shared/api/requests/leaders';
+import { addLeader } from 'redux/leader/reducer';
+import { setLeadersOfCurrentDay } from 'redux/history/reducer';
+
+const AddForm = ({ handleClose }) => {
+  const [values, setValues] = useState({ name: '', score: 0 });
+  const dispatch = useDispatch();
+  const { leaders, sortType } = useSelector(state => state.leader);
+  const addLeaderInList = leaderData => dispatch(addLeader(leaderData, sortType));
+
+  useEffect(() => {
+    dispatch(setLeadersOfCurrentDay(leaders));
+  }, [leaders]);
+
+  const handleChange = e => {
+    setValues({
+      ...values,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const onSubmit = async e => {
+    e.preventDefault();
+    const leaderResponse = await createLeader(values);
+    if (leaderResponse) {
+      addLeaderInList({
```

Comment:
create const payload = {
        ...leaderResponse,
        score: +leaderResponse.score,
        id: leaderResponse.id,
      }. and put to addLeaderInList(payload)